.github/workflows/test-windows.yml

```diff
@@ -0,0 +1,189 @@
+      - name: Cache conan packages
+        if: startsWith(matrix.os, 'windows')
+        uses: actions/cache@v3
+        with:
+          key: ${{ runner.os }}-conan-user-${{ hashFiles('**/conanfile.py') }}${{ env.C2_CONAN_CACHE_SUFFIX }}
+          path: ~/.conan2/
+
+      - name: Install Conan
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          python3 -c "import site; import sys; print(f'{site.USER_BASE}\\Python{sys.version_info.major}{sys.version_info.minor}\\Scripts')" >> "$Env:GITHUB_PATH"
+          pip3 install --user "conan==${{ env.CONAN_VERSION }}"
+        shell: pwsh
+
+      - name: Setup Conan
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          conan --version
+          conan profile detect -f
+
+      - name: Install dependencies
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          mkdir build-test
+          cd build-test
+          conan install .. `
+              -s build_type=RelWithDebInfo `
+              -c tools.cmake.cmaketoolchain:generator="NMake Makefiles" `
+              -b missing `
+              --output-folder=. `
+              -o with_openssl3="$Env:C2_USE_OPENSSL3"
+        shell: powershell
+
+      - name: Build
+        if: startsWith(matrix.os, 'windows')
+        shell: pwsh
+        env:
+          # Enable PCH on Windows, later we can have it depend on the run/Qt Version like in the build workflow, I just want to make it work
+          C2_WINDOWS_USE_PCH: True
+        run: |
+          cmake `
+              -G"NMake Makefiles" `
+              -DCMAKE_BUILD_TYPE=RelWithDebInfo `
+              -DBUILD_TESTS=On `
+              -DBUILD_APP=OFF `
+              -DCMAKE_TOOLCHAIN_FILE="conan_toolchain.cmake" `
+              -DUSE_PRECOMPILED_HEADERS=${{ env.C2_WINDOWS_USE_PCH }} `
+              -DBUILD_WITH_CRASHPAD="$Env:C2_ENABLE_CRASHPAD" `
+              -DCHATTERINO_LTO="$Env:C2_ENABLE_LTO" `
+              -DCHATTERINO_PLUGINS="$Env:C2_PLUGINS" `
+              -DBUILD_WITH_QT6="$Env:C2_BUILD_WITH_QT6" `
+              ..
+          set cl=/MP
+          nmake /S /NOLOGO
+        working-directory: build-test
+
+      - name: Download and extract Twitch PubSub Server Test
+        if: startsWith(matrix.os, 'windows')
+        run: |
+          mkdir pubsub-server-test
+          Invoke-WebRequest -Uri "https://github.com/Chatterino/twitch-pubsub-server-test/releases/download/v1.0.7/server-v1.0.7-windows-amd64.zip" -outfile "pubsub-server.zip"
+          Expand-Archive pubsub-server.zip -DestinationPath pubsub-server-test
+          rm pubsub-server.zip
+        shell: pwsh
+
+      - name: Checkout httpbox
+        if: startsWith(matrix.os, 'windows')
+        uses: actions/checkout@v4
+        with:
+          repository: "kevinastone/httpbox"
+          path: "httpbox"
+
+      - name: Test
+        if: startsWith(matrix.os, 'windows')
+        timeout-minutes: 30
+        run: |
+          cd ..\httpbox
+          cargo build
```

Comment:
You should be able to use `cargo install --git https://github.com/kevinastone/httpbox` (pretty sure this could accept a commit/tag as well). This should be in its own step to make sure it doesn't contribute to the 30 min. Maybe we should fork the project and add releases (with `cargo-dist` for example). Then we could use [`cargo-binstall`](https://github.com/cargo-bins/cargo-binstall)

Reply:
Just tested the following command locally: `cargo install --git https://github.com/kevinastone/httpbox --rev 89b971f` and it seemed to have worked. If a tag existed, I could use `--tag` instead of `--rev` according to `https://doc.rust-lang.org/cargo/commands/cargo-install.html`.
Regarding the separate step, I just did it in the same step because the docker containers in the ubuntu test were started in the Test step too.